app/src/main/scala/org/alephium/explorer/service/BlockFlowClient.scala

```diff
@@ -217,6 +240,83 @@ object BlockFlowClient extends StrictLogging {
     def fetchChainParams(): Future[ChainParams] =
       _send(getChainParams, uri, ())
 
+    def guessTokenStdInterfaceId(token: TokenId): Future[Option[StdInterfaceId]] = {
+      val address = Address.contract(ContractId.unsafe(token.value))
+      guessStdInterfaceId(address)
+    }
+
+    def guessStdInterfaceId(address: Address.Contract): Future[Option[StdInterfaceId]] = {
+      val group = address.groupIndex(groupSetting.groupConfig)
+      _send(contractState, uri, (address, group))
+        .map { rawState =>
+          rawState.immFields.lastOption match {
+            case Some(api.model.ValByteVec(bytes)) =>
+              val data = Hex.toHexString(bytes)
+              if (data.startsWith(interfaceIdPrefix)) {
+                Some(StdInterfaceId.from(data.drop(interfaceIdPrefix.size)))
+              } else {
+                Some(StdInterfaceId.NonStandard)
+              }
+            case _ => Some(StdInterfaceId.NonStandard)
```

Comment:
I think we should differentiate the two cases:
1. the last immFields starts with `interfaceIdPrefix`, but is not recongnized yet
2. the last immFields does not start with `interfaceIdPrefix`

Reply:
:+1: What about: 

1: `Unknown` and we store the id
2: `NonStandard`

Reply:
LGTM

Reply:
done